### PR TITLE
Add type annotations for test_reporting

### PR DESCRIPTION
**Pull request by Dhwaniartefact**

### CODE REVIEW

File: fixity/reporting.py [hunk 1]
```diff
@@ -51,7 +61,13 @@ def post_pre_scan_report(aip, start_time, report_url, report_auth=(), session_id
     return True
 
 
-def post_success_report(aip, report, report_url, report_auth=(), session_id=None):
+def post_success_report(
+    aip: str,
+    report: Optional[Report],
+    report_url: Optional[str],
```
Comment (replaceafill) [inline fixity/reporting.py]: Why are these `Optional`?

File: fixity/reporting.py [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
 
 class ReportServiceException(Exception):
     pass
 
 
-def post_pre_scan_report(aip, start_time, report_url, report_auth=(), session_id=None):
+def post_pre_scan_report(
+    aip: str,
+    start_time: Optional[datetime],
+    report_url: Optional[str],
```
Comment (replaceafill) [inline fixity/reporting.py]: Why are these `Optional`?

File: fixity/reporting.py (hunk 1 above)
Comment (Dhwaniartefact) [reply fixity/reporting.py]: Some tests are setting as report=None and report_url=None.

File: fixity/reporting.py (hunk 2 above)
Comment (Dhwaniartefact) [reply fixity/reporting.py]: Same answer as above.

File: fixity/reporting.py (hunk 1 above)
Comment (replaceafill) [reply fixity/reporting.py]: What do you think if we focus on the actual use from the `fixity` package and adjust the tests accordingly if needed?

File: fixity/reporting.py (hunk 2 above)
Comment (replaceafill) [reply fixity/reporting.py]: Same argument as above :grin:

File: fixity/reporting.py (hunk 1 above)
Comment (Dhwaniartefact) [reply fixity/reporting.py]: ya let me adjust the tests which focus on the actual use of the codes.

File: fixity/reporting.py (hunk 2 above)
Comment (Dhwaniartefact) [reply fixity/reporting.py]: Good one. :) I got your point. Let me adjust the tests.

File: fixity/reporting.py [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
 class ReportServiceException(Exception):
     pass
 
 
-def post_pre_scan_report(aip, start_time, report_url, report_auth=(), session_id=None):
+def post_pre_scan_report(
+    aip: str,
+    start_time: datetime,
+    report_url: str,
+    report_auth: str = (),
```
Comment (replaceafill) [inline fixity/reporting.py]: Why is this typed as `str` when it expects a `tuple`?

Comment (Dhwaniartefact) [reply fixity/reporting.py]: Aah yes, I missed to see [this](https://github.com/artefactual/fixity/blob/6eb4807a802dacee36beda99526ebddc05847817/fixity/fixity.py#L92-L96).

**Comment by Dhwaniartefact** [comment]
We will annotate `test_reporting` in future PRs.

**Pull request closed without merging**